src/rpc/mining.cpp

```diff
@@ -572,24 +580,36 @@ UniValue getblocktemplate(const UniValue& params, bool fHelp)
             nTransactionsUpdatedLastLP = nTransactionsUpdatedLast;
         }
 
-        // Release the wallet and main lock while waiting
-        LEAVE_CRITICAL_SECTION(cs_main);
         {
-            checktxtime = boost::get_system_time() + boost::posix_time::minutes(1);
```

Comment:
This one minute time is far too long for us, it's the same as current bitcoin (10 minute blocks); this should be much shorter than the target block interval. This time determines how soon we check for transactions to have arrived in the mempool.

I'm not sure 1 minute makes sense even for bitcoin, because block interval is a Poisson distribution. This seems to be assuming that a block won't be solved for a while, so wait 1 minute initially, but then start checking every 10 seconds. It probably should be 10 seconds even in bitcoin.

Reply:
This is a great catch